firedrake/mg/embedded.py

```diff
@@ -28,6 +28,9 @@ def __init__(self, element):
             degree = element.degree()
             family = lambda c: "DG" if c.is_simplex() else "DQ"
             if isinstance(cell, ufl.TensorProductCell):
+                if type(degree) is int:
+                    degree = [degree] * len(cell.sub_cells())
+
                 scalar_element = ufl.TensorProductElement(*(ufl.FiniteElement(family(c), cell=c, degree=d)
                                                             for (c, d) in zip(cell.sub_cells(), degree)))
```

Comment:
This code will produce the embedding DG element in `element.cell()` with `element.degree()`, regardless of `element.family()`. DG is a TPE if the cell is also TP. 
Serendipity can only be specified with an integer degree, so it might be better to construct DG via `TensorProductElement` if the degree is a tuple with different values, and we can use the standard `FiniteElement` constructor otherwise . 

Am I missing something? @rckirby any ideas?